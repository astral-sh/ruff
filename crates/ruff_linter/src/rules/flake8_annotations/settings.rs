//! Settings for the `flake-annotations` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Default, CacheKey)]
#[allow(clippy::struct_excessive_bools)]
pub struct Settings {
    pub mypy_init_return: bool,
    pub suppress_dummy_args: bool,
    pub suppress_none_returning: bool,
    pub allow_star_arg_any: bool,
    pub ignore_fully_untyped: bool,
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_annotations",
            fields = [
                self.mypy_init_return,
                self.suppress_dummy_args,
                self.suppress_none_returning,
                self.allow_star_arg_any,
                self.ignore_fully_untyped
            ]
        }
        Ok(())
    }
}
