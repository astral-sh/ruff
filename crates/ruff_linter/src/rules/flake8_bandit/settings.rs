//! Settings for the `flake8-bandit` plugin.

use ruff_macros::CacheKey;

pub fn default_tmp_dirs() -> Vec<String> {
    ["/tmp", "/var/tmp", "/dev/shm"]
        .map(ToString::to_string)
        .to_vec()
}

#[derive(Debug, CacheKey)]
pub struct Settings {
    pub hardcoded_tmp_directory: Vec<String>,
    pub check_typed_exception: bool,
}

impl Default for Settings {
    fn default() -> Self {
        Self {
            hardcoded_tmp_directory: default_tmp_dirs(),
            check_typed_exception: false,
        }
    }
}
