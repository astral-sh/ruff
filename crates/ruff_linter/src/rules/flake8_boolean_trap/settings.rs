//! Settings for the `flake8-boolean-trap` plugin.

use std::fmt;

use ruff_macros::CacheKey;

use crate::display_settings;

#[derive(Debug, Clone, CacheKey, Default)]
pub struct Settings {
    pub extend_allowed_calls: Vec<String>,
}

impl fmt::Display for Settings {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_boolean_trap",
            fields = [
                self.extend_allowed_calls | array,
            ]
        }
        Ok(())
    }
}
