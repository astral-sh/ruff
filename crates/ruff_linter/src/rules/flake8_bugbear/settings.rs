//! Settings for the `flake8-bugbear` plugin.

use ruff_macros::CacheKey;

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub extend_immutable_calls: Vec<String>,
}
