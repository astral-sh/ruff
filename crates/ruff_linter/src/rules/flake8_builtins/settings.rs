//! Settings for the `flake8-builtins` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub builtins_ignorelist: Vec<String>,
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_builtins",
            fields = [
                self.builtins_ignorelist | array
            ]
        }
        Ok(())
    }
}
