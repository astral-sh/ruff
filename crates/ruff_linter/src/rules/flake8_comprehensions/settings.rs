//! Settings for the `flake8-comprehensions` plugin.

use ruff_macros::CacheKey;

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub allow_dict_calls_with_keyword_arguments: bool,
}
