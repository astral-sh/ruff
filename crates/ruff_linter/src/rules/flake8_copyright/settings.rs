//! Settings for the `flake8-copyright` plugin.

use once_cell::sync::Lazy;
use regex::Regex;
use std::fmt::{Display, Formatter};

use crate::display_settings;
use ruff_macros::CacheKey;

#[derive(Debug, CacheKey)]
pub struct Settings {
    pub notice_rgx: Regex,
    pub author: Option<String>,
    pub min_file_size: usize,
}

pub static COPYRIGHT: Lazy<Regex> =
    Lazy::new(|| Regex::new(r"(?i)Copyright\s+(\(C\)\s+)?\d{4}(-\d{4})*").unwrap());

impl Default for Settings {
    fn default() -> Self {
        Self {
            notice_rgx: COPYRIGHT.clone(),
            author: None,
            min_file_size: 0,
        }
    }
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_copyright",
            fields = [
                self.notice_rgx,
                self.author | optional,
                self.min_file_size,
            ]
        }
        Ok(())
    }
}
