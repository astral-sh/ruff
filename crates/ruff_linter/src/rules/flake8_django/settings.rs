//! Settings for the `flake8-django` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Clone, CacheKey)]
pub struct Settings {
    pub additional_path_functions: Vec<String>,
}

impl Default for Settings {
    fn default() -> Self {
        Self {
            additional_path_functions: vec![],
        }
    }
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_django",
            fields = [
                self.additional_path_functions | array
            ]
        }
        Ok(())
    }
}
