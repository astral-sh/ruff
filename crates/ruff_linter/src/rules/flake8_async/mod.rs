//! Rules from [flake8-async](https://pypi.org/project/flake8-async/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::assert_messages;
    use crate::registry::Rule;
    use crate::settings::LinterSettings;
    use crate::test::test_path;

    #[test_case(Rule::BlockingHttpCallInAsyncFunction, Path::new("ASYNC100.py"))]
    #[test_case(Rule::OpenSleepOrSubprocessInAsyncFunction, Path::new("ASYNC101.py"))]
    #[test_case(Rule::BlockingOsCallInAsyncFunction, Path::new("ASYNC102.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.noqa_code(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("flake8_async").join(path).as_path(),
            &LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
