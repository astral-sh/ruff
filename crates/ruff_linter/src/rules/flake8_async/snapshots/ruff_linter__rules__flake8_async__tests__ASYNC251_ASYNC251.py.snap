---
source: crates/ruff_linter/src/rules/flake8_async/mod.rs
snapshot_kind: text
---
ASYNC251.py:6:5: ASYNC251 Async functions should not call `time.sleep`
  |
5 | async def func():
6 |     time.sleep(1)  # ASYNC251
  |     ^^^^^^^^^^ ASYNC251
  |
